crates/chia-consensus/src/test_generators.rs

```diff
@@ -369,43 +387,53 @@ fn run_generator(#[case] name: &str) {
-            for (i, spend) in conds.spends.iter().enumerate() {
-                let runnable = {
-                    let mut a = make_allocator(flags);
-                    coinspends[i]
-                        .puzzle_reveal
-                        .run(
-                            &mut a,
-                            flags,
-                            TEST_CONSTANTS.max_block_cost_clvm,
-                            &coinspends[i].solution,
-                        )
-                        .is_ok()
-                };
-                assert!(runnable);
-                let parent_id = a2.atom(spend.parent_id);
-                assert_eq!(
-                    parent_id.as_ref(),
-                    coinspends[i].coin.parent_coin_info.as_slice()
-                );
-                let puzhash = a2.atom(spend.puzzle_hash);
-                assert_eq!(puzhash.as_ref(), coinspends[i].coin.puzzle_hash.as_slice());
-                assert_eq!(spend.coin_amount, coinspends[i].coin.amount);
 
-                // check that we're getting the same info from get_coinspends_with_conditions_for_trusted_block
-                assert_eq!(
-                    parent_id.as_ref(),
-                    coinspends2[i].0.coin.parent_coin_info.as_slice()
-                );
-                assert_eq!(
-                    puzhash.as_ref(),
-                    coinspends2[i].0.coin.puzzle_hash.as_slice()
-                );
-                assert_eq!(spend.coin_amount, coinspends2[i].0.coin.amount);
+            // get_coinspends_for_trusted_block skips certain attacking spends as it is required to
+            // return serialized puzzles, which may not be possible. We can ignore the comparison
+            // if it has skipped a spend for this reason
+            if coinspends.len() == conds.spends.len() {
+                for (i, spend) in conds.spends.iter().enumerate() {
+                    let runnable = {
+                        let mut a = make_allocator(flags);
+                        coinspends[i]
+                            .puzzle_reveal
+                            .run(
+                                &mut a,
+                                flags,
+                                TEST_CONSTANTS.max_block_cost_clvm,
+                                &coinspends[i].solution,
+                            )
+                            .is_ok()
+                    };
+                    assert!(runnable);
+                    let parent_id = a2.atom(spend.parent_id);
+                    assert_eq!(
+                        parent_id.as_ref(),
+                        coinspends[i].coin.parent_coin_info.as_slice()
+                    );
+                    let puzhash = a2.atom(spend.puzzle_hash);
+                    assert_eq!(puzhash.as_ref(), coinspends[i].coin.puzzle_hash.as_slice());
+                    assert_eq!(spend.coin_amount, coinspends[i].coin.amount);
+
+                    // test with_conditions
```

Comment:
```suggestion
                    // test get_coinspends_with_conditions_for_trusted_block()
```
Is that what you meant?